### PR TITLE
New Extension: WakaTime

### PR DESCRIPTION
**Pull request by iammola**
## Description

It's an extension to track your WakaTime stats.

## Screencast
I couldn't get a screenshot to fit the specs without scaling it. I know the guidelines said I shouldn't and that's why I didn't put it in the metadata folder. I noticed some other extensions don't have screenshots as well, and I thought it'll be fine to submit the PR without one at first. I'll appreciate any help I can get with this. Anyways, Here are the ones I could take.

#### Leaderboard
leaderboard

#### Summary
summary

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by koinzhang** [comment]
Hi @iammola 
Try this extension [Capture Raycast Metadata](https://www.raycast.com/koinzhang/capture-raycast-metadata)
With the help of this you can easily take the screenshots required for the metadata
Please give Raycast access to screen recording in the system settings first.

**Comment by iammola** [comment]
Hey @koinzhang.

I just tried the extension and it works fine. Except the screenshot size's is 1000x625. I already tried changing my screen resolution using SwitchResX. But everyone I use zooms out the entire desktop and I don't have the option to increase my text size like I've seen on other macOS devices.

leaderboard

The toast shows up too early as well. So I guess that's a bug. But regardless, if I had a good screen resolution to take the screenshots as required, I'll definitely use your extension

File: extensions/wakatime/src/leaderboard.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export default function Command({ id }: { id?: string }) {
+  const { data, isLoading } = useLeaderBoard(id);
+  const [showDetail, setShowDetail] = useState(false);
+
+  return (
+    <List
+      isLoading={isLoading}
+      isShowingDetail={showDetail}
+      selectedItemId={data?.current_user.user.id}
+      navigationTitle={data !== undefined ? `Page ${data?.page} of ${data?.total_pages}` : ""}
```
Comment (pernielsentikaer) [inline extensions/wakatime/src/leaderboard.tsx:16]:
```suggestion
```
Don't change the navigationTitle in the root command.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @iammola 👋 

Thanks for your contribution 🔥 

I have some comments and suggestions for you to address:

Looking at this:
CleanShot 2022-05-04 at 08 57 25@2x
Both hours and stats seems weird (can be because of an interval)

Is there something wrong with the numbers?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by iammola** [comment]
> Hi @iammola 👋
> 
> Thanks for your contribution 🔥
> 
> I have some comments and suggestions for you to address:
> 
> Looking at this: CleanShot 2022-05-04 at 08 57 25@2x Both hours and stats seems weird (can be because of an interval)
> 
> Is there something wrong with the numbers?
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

I just used the title WakaTIme used to name their columns from [https://wakatime.com/leaders](https://wakatime.com/leaders). That's where the public leaderboard is. I assume it's the amount of time-coded over the last 7 days from yesterday.

WakaTime Leaders Info

I could rename the column or actually get their all-time coding hours?

**Review by pernielsentikaer — APPROVED** [review]
Hi @iammola 👋 

Thanks for the extension; already looking forward to what you're going to add next time 😄

**Pull request merged**